waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -245,6 +250,14 @@ constructAppSpec waspDir options externalConfigs parsedPrismaSchema decls = do
 
   return $ runValidation ASV.validateAppSpec appSpec
 
+waspDirExists :: Path' Abs (Dir WaspProjectDir) -> IO (Either String (Path' Abs (Dir WaspProjectDir)))
+waspDirExists waspDir = do
```

Comment:
This function is called `waspDirExists`, implying that it checks whether a `.wasp` directory exists and returns either an error or a path to the directory.

But, in reality, the function does something else:
- If a `.wasp` directory does exist, it returns a path to the existing `.wasp` directory (so far so good)
- If the `.wasp` directory doesn't exist, it returns a path to the non-existing `.wasp` directory (this is the first weird part)
- If a `.wasp` file exists, it returns an error saying that `.wasp` is a file (but the function's name implies it should only care about whether the `.wasp` directory exists and report on that).

Imagine what happens when someone else tries to use this function in a different context (e.g., to really check whether the `.wasp` directory exists). I think they'd end up very confused :)

Anyway, for this issue, we don't care whether the `.wasp` directory exists or not. It can exist, it doesn't need to exist - it's all the same to us.

We only care about whether a `.wasp` file exists, so we should write a function to check for that, and name it accordingly.

Reply:
My original plan for this implementation was to create a sort of pipeline in which the result of the previous function would go into the next function, as it is usually done in [railway oriented programming](https://blog.logrocket.com/what-is-railway-oriented-programming/). 

This way, I would be able to avoid having two `case` statements in the `analyzeWaspProject`, because the next `waspFilePathOrError` would only be called if its input was a *right* value, otherwise it will pass through the *left* value unchanged.

Unfortunately, I was unable to do that, and I forgot to change the return type of this function to `boolean` or an empty *right* value.